tests/gemm.cpp

```diff
@@ -369,4 +370,394 @@ TEST(BLAS3_trmm, left_lower_tri_implicit_diag)
+  {
+    /* C = B * A, where A is triangular mxm */
+    using extents_t = extents<dynamic_extent, dynamic_extent>;
+    using matrix_t = mdspan<double, extents_t, layout_left>;
+    double snan = std::numeric_limits<double>::signaling_NaN();
+
+    int m = 3, n = 2;
+    std::vector<double> A_mem(m*m, snan);
+    std::vector<double> B_mem(m*n);
+    std::vector<double> C_mem(m*n, snan);
+    std::vector<double> gs_mem(m*n);
+
+    matrix_t A(A_mem.data(), m, m);
+    matrix_t B(B_mem.data(), n, m);
+    matrix_t C(C_mem.data(), n, m);
+    matrix_t gs(gs_mem.data(), n, m);
+
+    // Fill A
+    A(0,0) = 3.5;
+    A(1,0) = -2.0;
+    A(1,1) = 1.2;
+    A(2,0) = -0.1;
+    A(2,1) = 4.5;
+    A(2,2) = -1.0;
+    
+    // Fill B
+    B(0,0) = -4.4;
+    B(1,0) = 1.8;
+    B(0,1) = -1.4;
+    B(1,1) = 3.4;
+    B(0,2) = 1.8;
+    B(1,2) = 1.6;
+
+    // Fill GS
+    gs(0,0) = -12.78;
+    gs(1,0) = -0.66;
+    gs(0,1) = 6.42;
+    gs(1,1) = 11.28;
+    gs(0,2) = -1.8;
+    gs(1,2) = -1.6;
+
+    // Check the non-overwriting version
+    triangular_matrix_right_product(A, lower_triangle, explicit_diagonal, B, C);
+
+    for (ptrdiff_t j = 0; j < m; ++j) {
+      for (ptrdiff_t i = 0; i < n; ++i) {
+        EXPECT_DOUBLE_EQ(gs(i,j), C(i,j)) 
+          << "Matrices differ at index (" 
+          << i << "," << j << ")\n";
+      }
+    }
+
+    // Check the overwriting version
+    triangular_matrix_right_product(A, lower_triangle, explicit_diagonal, B);
+
+    for (ptrdiff_t j = 0; j < m; ++j) {
+      for (ptrdiff_t i = 0; i < n; ++i) {
+        EXPECT_DOUBLE_EQ(gs(i,j), B(i,j)) 
+          << "Matrices differ at index (" 
+          << i << "," << j << ")\n";
+      }
+    }
+  }
+
+TEST(BLAS3_trmm, right_lower_tri_implicit_diag)
+  {
+    /* C = A * B, where A is triangular mxm */
+    using extents_t = extents<dynamic_extent, dynamic_extent>;
+    using matrix_t = mdspan<double, extents_t, layout_left>;
+    double snan = std::numeric_limits<double>::signaling_NaN();
+
+    int m = 3, n = 2;
+    std::vector<double> A_mem(m*m, snan);
+    std::vector<double> B_mem(m*n);
+    std::vector<double> C_mem(m*n, snan);
+    std::vector<double> gs_mem(m*n);
+
+    matrix_t A(A_mem.data(), m, m);
+    matrix_t B(B_mem.data(), n, m);
+    matrix_t C(C_mem.data(), n, m);
+    matrix_t gs(gs_mem.data(), n, m);
+
+    // Fill A
+    A(1,0) = -2.0;
+    A(2,0) = -0.1;
+    A(2,1) = 4.5;
+    
+    // Fill B
+    B(0,0) = -4.4;
+    B(1,0) = 1.8;
+    B(0,1) = -1.4;
+    B(1,1) = 3.4;
+    B(0,2) = 1.8;
+    B(1,2) = 1.6;
+
+    triangular_matrix_right_product(A, lower_triangle, implicit_unit_diagonal, B, C);
+
+    // Fill GS
+    gs(0,0) = -1.78;
+    gs(1,0) = -5.16;
+    gs(0,1) = 6.7;
+    gs(1,1) = 10.6;
+    gs(0,2) = 1.8;
+    gs(1,2) = 1.6;
+
+    for (ptrdiff_t j = 0; j < m; ++j) {
+      for (ptrdiff_t i = 0; i < n; ++i) {
+        // FIXME: Choose a more reasonable value for the tolerance
+        double tol = 1e-9;
```

Comment:
Would you consider making constants like these `constexpr` if possible?  Thanks!

Reply:
Done!